src/runtime/TemporalObject.cpp

```diff
@@ -102,19 +102,33 @@ std::map<TemporalObject::DateTimeUnits, int> TemporalObject::getSeconds(Executio
              { NANOSECOND_UNIT, std::stoi(tmp.substr(6, 3)) } };
 }
 
-void TemporalObject::offset(ExecutionState& state, std::string& isoString, unsigned int& index)
+std::string TemporalObject::offset(ExecutionState& state, std::string& isoString, unsigned int& index)
```

Comment:
offset is a string not a number?

Reply:
It just validates if a string is valid `offsetString`. If it's valid then returns it. An other function will process the offsetString to nanoseconds. Not always needed to process it to number.